mp/src/game/server/momentum/mom_triggers.cpp

```diff
@@ -1560,26 +1561,38 @@ void CNoGrenadesZone::Precache()
 
 bool CNoGrenadesZone::IsInsideNoGrenadesZone(CBaseEntity *pOther)
 {
-    if ( pOther )
+    if (!pOther)
+        return false;
+
+    auto pSticky = dynamic_cast<CMomStickybomb *>(pOther);
+    FOR_EACH_VEC(s_vecNoGrenadeZones, i)
     {
-        const auto pSticky = dynamic_cast<CMomStickybomb*>(pOther);
-        FOR_EACH_VEC(s_vecNoGrenadeZones, i)
-        {
-            const auto pNoGrenadeZone = s_vecNoGrenadeZones[i];
+        const auto pNoGrenadeZone = s_vecNoGrenadeZones[i];
 
-            if (pNoGrenadeZone->m_bDisabled || !pNoGrenadeZone->PointIsWithin(pOther->GetAbsOrigin()))
-                continue;
+        if (pNoGrenadeZone->m_bDisabled || !pNoGrenadeZone->PointIsWithin(pOther->GetAbsOrigin()))
+            continue;
 
-            if (!pSticky)
-                return true; // This is a rocket in a nonade zone and should fizzle
+        if (!pSticky)
+            return true; // This is a rocket in a nonade zone and should fizzle
 
-            return !pNoGrenadeZone->m_bAirborneOnly || !pSticky->DidHitWorld();
-        }
+        return pNoGrenadeZone->m_iStickyFizzleType == FizzleOnLand || pSticky->IsArmed() && (!pNoGrenadeZone->m_bAirborneOnly || !pSticky->DidHitWorld());
```

Comment:
The arm state of the sticky shouldn't matter here. Still, with this logic stickies fizzle when hitting the ground and `airborne_only` is `true`.